src/platform/imx8ulp/include/platform/lib/memory.h

```diff
@@ -0,0 +1,220 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright 2021 NXP
+ *
+ * Zhang Peng <peng.zhang_8@nxp.com>
+ */
+
+#ifdef __SOF_LIB_MEMORY_H__
+
+#ifndef __PLATFORM_LIB_MEMORY_H__
+#define __PLATFORM_LIB_MEMORY_H__
+
+#include <sof/lib/cache.h>
+
+
+/* data cache line alignment */
+#define PLATFORM_DCACHE_ALIGN	sizeof(void *)
+
+/* physical DSP addresses */
+
+#define IRAM_BASE	0x21170000
+#define IRAM_SIZE	0x10000
+
+#define DRAM0_BASE	0x21180000
+#define DRAM0_SIZE	0x10000
+
+#define SDRAM0_BASE	0x1a000000
+#define SDRAM0_SIZE	0x800000
+
+#define SDRAM1_BASE	0x1a800000
+#define SDRAM1_SIZE	0x800000
+
+/* physical address in Arm side */
+#define MEM_RESERVED    0x8e000000
+
+/* MU3: ASide (A35) BSide (HIFI4) */
+#define XSHAL_MU3_SIDEB_BYPASS_PADDR 0x2DA20000
+#define MU_BASE		XSHAL_MU3_SIDEB_BYPASS_PADDR
+
+#define EDMA2_BASE	0x2D810000
+#define EDMA2_SIZE	0x10000
+
+#define SAI_5_BASE	0x29890000
+#define SAI_5_SIZE	0x00010000
+#define SAI_6_BASE	0x2DA90000
+#define SAI_6_SIZE	0x00010000
+#define SAI_7_BASE	0x2DAA0000
+#define SAI_7_SIZE	0x00010000
+
+#define UUID_ENTRY_ELF_BASE	0x1FFFA000
+#define UUID_ENTRY_ELF_SIZE	0x6000
+
+#define LOG_ENTRY_ELF_BASE	0x20000000
+#define LOG_ENTRY_ELF_SIZE	0x2000000
+
+#define EXT_MANIFEST_ELF_BASE	(LOG_ENTRY_ELF_BASE + LOG_ENTRY_ELF_SIZE)
+#define EXT_MANIFEST_ELF_SIZE	0x2000000
+
+/*
+ * The Heap and Stack on i.MX8 are organised like this :-
+ *
+ * +--------------------------------------------------------------------------+
+ * | Offset              | Region         |  Size                             |
+ * +---------------------+----------------+-----------------------------------+
+ * | SDRAM_BASE          | RO Data        |  SOF_DATA_SIZE                    |
+ * |                     | Data           |                                   |
+ * |                     | BSS            |                                   |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_SYSTEM_BASE    | System Heap    |  HEAP_SYSTEM_SIZE                 |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_RUNTIME_BASE   | Runtime Heap   |  HEAP_RUNTIME_SIZE                |
+ * +---------------------+----------------+-----------------------------------+
+ * | HEAP_BUFFER_BASE    | Module Buffers |  HEAP_BUFFER_SIZE                 |
+ * +---------------------+----------------+-----------------------------------+
+ * | SOF_STACK_END       | Stack          |  SOF_STACK_SIZE                   |
+ * +---------------------+----------------+-----------------------------------+
+ * | SOF_STACK_BASE      |                |                                   |
+ * +---------------------+----------------+-----------------------------------+
+ */
+
+/* Mailbox configuration */
+#define SRAM_OUTBOX_BASE	SDRAM1_BASE
+#define SRAM_OUTBOX_SIZE	0x1000
+#define SRAM_OUTBOX_OFFSET	0
+
+#define SRAM_INBOX_BASE		(SRAM_OUTBOX_BASE + SRAM_OUTBOX_SIZE)
```

Comment:
Can you please try to align the values?

Reply:
Looks good if you fetch the PR and look at it with an editor.